lib/btree.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+const DEFAULT_DEGREE = 6; // min degree of b-tree.
+// All vertices except the root have [degree ... 2 * degree] child nodes
+// And [degree - 1 ... 2 * degree - 1] + 1 "empty" elements
+
+class Element {
+  constructor(value, tuple, less = null) {
+    this.value = value;
+    this.tuple = tuple;
+    this.less = less;
+  }
+}
+
+const empty = less => new Element(undefined, null, less);
+
+const destroyNode = (parent, index) => {
+  const node = parent[index].less;
+  const len = node.length;
+  const newLeftNode = node.splice(0, len / 2 - 1); // First half
+  const mid = node.shift();
+  newLeftNode.push(empty(mid.less));
+  mid.less = newLeftNode;
+  parent.splice(index, 0, mid);
+  return parent;
+};
+
+const isLeaf = node => !node[0].less;
+
+const binarySearch = (node, value) => {
+  let start = 0;
+  let end = node.length - 1;
+  while (start <= end) {
+    const i = (start + end) >> 1;
+    const item = node[i];
+    const itemValue = item.value;
+    if (value > itemValue) start = i + 1;
+    else if (!(value >= itemValue)) end = i - 1;
+    // undefined must be bigger then other value
+    // (!(1 >= undefined)) -> true, but just (1 < undefined) -> false
+    else return [true, item.tuple];
```

Comment:
I'd suggest returning at least an item itself (I assume you optimized it this way). Though I'd prefer to always return an integer (index) from this function this will make it both easier to understand and easier to optimize for the compiler (less type diversity, only integers).